pseudopotential_calculator/calculations/slab.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING, Self
+
+from ase.constraints import FixAtoms
+
+from pseudopotential_calculator.castep import CastepConfig, get_default_calculator
+
+if TYPE_CHECKING:
+    from ase import Atoms
+    from ase.calculators.castep import Castep
+
+    from pseudopotential_calculator.calculations.bulk import XCFunctional
+
+
+@dataclass
+class SlabOptimizationParams:
+    """Parameters of a slab calculation."""
```

Comment:
I think we should make some generic config to use for both slab and bulk, since the only difference is kpoints

Reply:
true